### ISSUE TITLE
Dumping memory signature

### ISSUE THREAD

**Issue by abdullahyildiz** [riscv/riscv-isa-sim#1037]
Does the [release v1.1.0](https://github.com/riscv-software-src/riscv-isa-sim/releases/tag/v1.1.0) support dumping memory signature?

There is no such option in this version of Spike binary:

https://github.com/riscv-software-src/riscv-isa-sim/blob/530af85d83781a3dae31a4ace84a573ec255fefa/spike_main/spike.cc#L18-L77

However, the host interface seems to support it:
https://github.com/riscv-software-src/riscv-isa-sim/blob/1767a27ad4cf8f9eb1a30d9d9a2495477071339f/fesvr/htif.cc#L150-L155

**Comment by Phantom1003** [comment]
Yes, memory signature still works, use `+signature=` to specify the signature file.
htif will also parse command line arguments, it looks like `-h` conflicts with spike, you can use `+h` instead.
https://github.com/riscv-software-src/riscv-isa-sim/blob/2a7c33b8b50f91379a12f983dd3c1c7ee9f89a90/spike_main/spike.cc#L408-L409
https://github.com/riscv-software-src/riscv-isa-sim/blob/1767a27ad4cf8f9eb1a30d9d9a2495477071339f/fesvr/htif.cc#L341-L344

**Comment by abdullahyildiz** [comment]
@Phantom1003,

Could you share the command syntax supporting `+signature=`?

**Comment by Phantom1003** [comment]
Try `spike +signature=<signature file> <testcase file>`, and use `spike +h` to get the help message from htif.

**Comment by abdullahyildiz** [comment]
There are other options that I need to add to the command line and the following one didn't work:

`spike -l --log-commits -m0x40000000:0x100000 --extlib=../spike_mmio_device_plugin/plugin.so --device=test_mmio_plugin,0x20000000,argument +signature=test.sig test.riscv`

**Comment by hakan-demirli** [comment]
I have encountered a similar issue. Spike either didn't do anything or complained:  ```unknown option "+signature"```. The problem was the position of the "+signature" and "--signature" arguments. Putting them right before the *.riscv file solved the issue.

This one works:
```
spike -l -m0x40000000:0x80000 --isa=rv32imc +signature=test_0178.sig --signature=test_0178.sig bin/test_0178.riscv 2> log/test_0178.log
```
This one works too:
```
spike -l -m0x40000000:0x80000 --isa=rv32imc +signature=test_0178.sig bin/test_0178.riscv 2> log/test_0178.log
```

This one does not:
```
spike -l -m0x40000000:0x80000 --isa=rv32imc --signature=test_0178.sig bin/test_0178.riscv 2> log/test_0178.log
```

As an alternative it is possible to dump the signatures via interactive mode.

- Add ```-d``` tag to the beginning of the command.
- ```run``` the code.
-  ```dump``` the whole memory. 
- Convert the binary file to ASCII Hex. 
- Since the beginning of the begin_signature section is known from the gcc dump file and it is aligned to 4 bytes. It can be extracted by directly copying the respective lines.
